src/plugins/data_views/server/fetcher/index_patterns_fetcher.ts

```diff
@@ -72,6 +73,9 @@ export class IndexPatternsFetcher {
     // if only one pattern, don't bother with validation. We let getFieldCapabilities fail if the single pattern is bad regardless
     if (patternList.length > 1 && !allowNoIndices) {
       patternListActive = await this.validatePatternListActive(patternList);
+      if (patternListActive.length === 0) {
+        throw new DataViewMissingIndices(patternList.join(','));
+      }
     }
```

Comment:
Since Matt is on PTO and also has been working on this issue, could you provide some context about this change, what is being fixed, and how it can be tested? Many thx!

BTW: some recent change in this area, to raise awareness https://github.com/elastic/kibana/pull/151788